packages/sdk-v2/src/sdk/onboard/modules/customwalletconnect/EthProvider.ts

```diff
@@ -0,0 +1,204 @@
+import WalletConnect from '@walletconnect/client'
+import { Subject, fromEvent } from 'rxjs'
+import { takeUntil, take } from 'rxjs/operators'
+import QRCodeModal from '@walletconnect/qrcode-modal'
+import { isMobile } from './helpers/isMobile'
+import { ProviderRpcError, ProviderRpcErrorCode } from '@web3-onboard/common'
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import EventEmitter from 'eventemitter3'
+import { AsyncStorage } from '@gooddollar/web3sdk-v2'
+
+import { WcConnectOptions } from './types'
+import type { Chain, EIP1193Provider, ProviderAccounts } from '@web3-onboard/common'
+import type { StaticJsonRpcProvider as StaticJsonRpcProviderType } from '@ethersproject/providers'
+
+export class EthProvider {
```

Comment:
For torus this approach works since we only need to update label/icon
for wallet-connect, 
I don't believe we can only overwrite the request method when method = 'eth_requestAccounts'
Only if we could access the connector initialized in the getInterface (is there a way?)

as far as I can tell, we now need to initialize a separate connector, 
which would end up with having to implement all the other stuff as well and that kind of defeats the whole purpose no?

Reply:
@L03TJ3 ^^^

Reply:
```javascript
provider.request = async function ({ method, params }: { method: string; params?: any }) {